internal/provider/resource/secret_sync/secret_sync_gitlab.go

```diff
@@ -0,0 +1,313 @@
+package resource
+
+import (
+	"context"
+	infisical "terraform-provider-infisical/internal/client"
+	infisicalclient "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+type SecretSyncGitlabDestinationConfigModel struct {
+	Scope                types.String `tfsdk:"scope"`
+	ProjectId            types.String `tfsdk:"project_id"`
+	ProjectName          types.String `tfsdk:"project_name"`
+	TargetEnvironment    types.String `tfsdk:"target_environment"`
+	ShouldProtectSecrets types.Bool   `tfsdk:"should_protect_secrets"`
+	ShouldMaskSecrets    types.Bool   `tfsdk:"should_mask_secrets"`
+	ShouldHideSecrets    types.Bool   `tfsdk:"should_hide_secrets"`
+}
+
+type SecretSyncGitlabSyncOptionsModel struct {
+	InitialSyncBehavior   types.String `tfsdk:"initial_sync_behavior"`
+	DisableSecretDeletion types.Bool   `tfsdk:"disable_secret_deletion"`
+	KeySchema             types.String `tfsdk:"key_schema"`
+}
+
+func NewSecretSyncGitlabResource() resource.Resource {
+	return &SecretSyncBaseResource{
+		App:              infisical.SecretSyncAppGitlab,
+		SyncName:         "GitLab",
+		ResourceTypeName: "_secret_sync_gitlab",
+		AppConnection:    infisical.AppConnectionAppGitlab,
+		DestinationConfigAttributes: map[string]schema.Attribute{
+			"scope": schema.StringAttribute{
+				Required:    true,
+				Description: "The scope to sync the secrets to. Must be 'project' for GitLab project variables.",
+			},
```

Comment:
we also support "group" right? 
<img width="715" height="751" alt="image" src="https://github.com/user-attachments/assets/58af55c9-03ce-4d4a-8bf1-931275fd524c" />